### ISSUE TITLE
Enable conv_fused/blas_compare/blas_compare_setup typecheck during CI

### ISSUE THREAD

**Issue by guilhermeleobas** [pytorch/pytorch#51234]
Enable typecheck for the following files:
- torch._overrides
- torch.utils.tensorboard._caffe2_graph
- torch.contrib._tensorboard_vis
- torch.nn.utils.prune
- torch.utils.show_pickle
- torch.utils.hipify.hipify_python
- torch.utils.benchmark.examples.*
- torch.nn.quantized.modules.batchnorm
- torch.nn.intrinsic.qat.modules.conv_fused

cc @ezyang @malfet @rgommers @xuzhao9 @gramster
